Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISSumBanks.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid.api import (AlgorithmFactory, DataProcessorAlgorithm, MatrixWorkspaceProperty, MatrixWorkspace, PropertyMode)
+from mantid.geometry import RectangularDetector
+from mantid.kernel import Direction
+
+
+class ReflectometryISISSumBanks(DataProcessorAlgorithm):
+    _WORKSPACE = 'InputWorkspace'
+    _ROI = 'ROIDetectorIDs'
+    _OUTPUT_WS = 'OutputWorkspace'
+
+    def __init__(self):
+        """Initialize an instance of the algorithm."""
+        DataProcessorAlgorithm.__init__(self)
+
+    def category(self):
+        """Return the categories of the algorithm."""
+        return 'Reflectometry\\ISIS;Workflow\\Reflectometry'
+
+    def name(self):
+        """Return the name of the algorithm."""
+        return 'ReflectometryISISSumBanks'
+
+    def summary(self):
+        """Return a summary of the algorithm."""
+        return "Sum banks from an ISIS reflectometry 2D detector into a single bank"
+
+    def seeAlso(self):
+        """Return a list of related algorithm names."""
+        return ['ReflectometryISISLoadAndProcess']
+
+    def PyInit(self):
+        self.declareProperty(
+            MatrixWorkspaceProperty(self._WORKSPACE, '', direction=Direction.Input, optional=PropertyMode.Mandatory),
+            doc='An input 2D detector workspace')
+        self.declareProperty(self._ROI, defaultValue="",
+                             doc="List of detector IDs to include")
+        self.declareProperty(
+            MatrixWorkspaceProperty(self._OUTPUT_WS, '', direction=Direction.Output),
+            doc='The preprocessed output workspace. If multiple input runs are specified '
+                'they will be summed into a single output workspace.')
+
+    def PyExec(self):
+        input_workspace = self.getProperty(self._WORKSPACE).value
+        roi_detector_ids = self.getProperty(self._ROI).value
+
+        masked_workspace = self.mask_detectors(input_workspace, roi_detector_ids)
+        summed_workspace = self.sum_banks(masked_workspace)
+
+        self.setProperty(self._OUTPUT_WS, summed_workspace)
+
+    def validateInputs(self):
+        issues = dict()
+        workspace = self.getProperty(self._WORKSPACE).value
+        try:
```

Comment:
Do we need to check for an empty dict here?

Reply:
I'm not sure what you mean? Isn't it supposed to return an empty dict if there aren't any errors?

Reply:
Sorry to be clear I meant if the list of ROI detectors is empty we should add an error to validate inputs
But I got the dictionary we return and the list we get in swapped when I wrote the comment